integration/angular/simple-message.ts

```diff
@@ -48,7 +48,9 @@ export const SimpleMessage = {
 
   toJSON(message: SimpleMessage): unknown {
     const obj: any = {};
-    message.numberField !== undefined && (obj.numberField = Math.round(message.numberField));
+    if (message.numberField !== 0) {
```

Comment:
Hi @haines ; just sanity checking, but you think we're good to drop the `!== undefined` check from before?

I know the `numberField` is typed as `number` anyway, so the check was being overly cautious, but I'm thinking we explicitly added it b/c of concerns some prior user/contributor had...

Dunno, personally I'm fine dropping it without a strong opinion / test case asserting why we should keep it, but just wanted to see if you had specific thoughts/strong/soft opinions on it. Wdyt?

Reply:
Yeah, personally I think it makes sense to drop it.

As far as I can tell the only reason this check would be necessary is if there is a bug in the calling code that circumvents the type system. `fromJSON` correctly populates the default value, and we have `fromPartial` for manually constructing messages without having to specify every field.

I'm not convinced it is worthwhile to guard against bugs in the calling code. We don't have other runtime type checks (e.g. checking that the values that are supposed to be numbers aren't actually strings), so should this be any different?

There was a test case [dating back to 2019](https://github.com/stephenh/ts-proto/commit/666a96c789b2b2eeaa4c0db77b7b9d83644c8461#diff-27827f36c0c9cddb97f98e7a65dc3495360678cde9dc87e0964e3a87957cb4dc) that exercised the checks by doing `Simple.toJSON({} as Simple)`, but I really don't think this is a valid use case (the type cast is incorrect) so [I removed it](https://github.com/stephenh/ts-proto/pull/878#discussion_r1262744555).

`useOptionals` is there when you want to allow things to be `undefined` in general, otherwise it is straightforward to call `fromPartial` before calling `toJSON` if you somehow have an object without the default values populated.